### DISCUSSION TITLE
Throwing error from Rust code (module) causes "error object is not a string"

### DISCUSSION THREAD

**Discussion by mauricelam** [mlua-rs/mlua#270] [Q&A]
How do I throw an error in Rust so that it will be printed out when called from lua?

Here's my test case for the behavior:

```rs
use mlua::prelude::*;

#[mlua::lua_module]
fn luaerror(_lua: &Lua) -> LuaResult<LuaTable> {
    Err("MyError".into_lua_err())
}
```

I am tried both 0.9.0-beta.1 and 0.8.8 on Mac using lua 5.1. My lua file is literally just `require('luaerror')` and the output I get is

```sh
$ cargo build && lua5.1 test.lua
lua5.1: (error object is not a string)
```

**Comment by khvzak** [comment]
I'm afraid lua 5.1 does not call `__tostring` metamethod when printing exceptions.
Normally you can print error by catching it and using [`tostring`](https://www.lua.org/manual/5.1/manual.html#pdf-tostring) function.
